lineoperations/src/lineoperations.c

```diff
@@ -0,0 +1,205 @@
+ *
+ *      You should have received a copy of the GNU General Public License along
+ *      with this program; if not, write to the Free Software Foundation, Inc.,
+ *      51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+
+#include "geanyplugin.h"
+#include "Scintilla.h"
+#include "linefunctions.h"
+
+
+static GtkWidget *main_menu_item = NULL;
+
+
+/* Remove Duplicate Lines, sorted */
+static void
+action_rmdupst_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmdupst(doc);
+}
+
+
+/* Remove Duplicate Lines, ordered */
+static void
+action_rmdupln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmdupln(doc);
+}
+
+
+/* Remove Unique Lines */
+static void
+action_rmunqln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmunqln(doc);
+}
+
+
+/* Remove Empty Lines */
+static void
+action_rmemtyln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmemtyln(doc);
+}
+
+
+/* Remove Whitespace Lines */
+static void
+action_rmwhspln_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) rmwhspln(doc);
+}
+
+
+/* Sort Lines Ascending */
+static void
+action_sortasc_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) sortlines(doc, 1);
+}
+
+
+/* Sort Lines Descending */
+static void
+action_sortdesc_item(GtkMenuItem *menuitem, gpointer gdata)
+{
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if(doc) sortlines(doc, 0);
+}
+
+
+static gboolean
+lo_init(GeanyPlugin *plugin, gpointer gdata)
+{
+	GeanyData *geany_data = plugin->geany_data;
+
+
+	GtkWidget *submenu;
+	GtkWidget *sep1;
+	GtkWidget *sep2;
+	GtkWidget *rmdupst_item;
+	GtkWidget *rmdupln_item;
+	GtkWidget *rmunqln_item;
+	GtkWidget *rmemtyln_item;
+	GtkWidget *rmwhspln_item;
+	GtkWidget *sortasc_item;
+	GtkWidget *sortdesc_item;
+
+	main_menu_item = gtk_menu_item_new_with_mnemonic(_("_Line Operations"));
+	gtk_widget_show(main_menu_item);
+
+	submenu = gtk_menu_new();
+	gtk_widget_show(submenu);
+	sep1 = gtk_separator_menu_item_new();
+	sep2 = gtk_separator_menu_item_new();
+	rmdupst_item  = gtk_menu_item_new_with_mnemonic(_("Remove Duplicate Lines, _Sorted"));
+	rmdupln_item  = gtk_menu_item_new_with_mnemonic(_("Remove Duplicate Lines, _Ordered"));
+	rmunqln_item  = gtk_menu_item_new_with_mnemonic(_("Remove _Unique Lines"));
+	rmemtyln_item = gtk_menu_item_new_with_mnemonic(_("Remove _Empty Lines"));
+	rmwhspln_item = gtk_menu_item_new_with_mnemonic(_("Remove _Whitespace Lines"));
+	sortasc_item  = gtk_menu_item_new_with_mnemonic(_("Sort Lines _Ascending"));
+	sortdesc_item = gtk_menu_item_new_with_mnemonic(_("Sort Lines _Descending"));
```

Comment:
Done in https://github.com/b4n/geany-plugins/commit/de106a6b044d8296858e1559a7b402fa8424903f if you're interested.  It doesn't depend on altering the callbacks so can be applied directly.